server/auth/sso/sso.go

```diff
@@ -280,6 +283,20 @@ func (s *sso) HandleCallback(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 	}
+	if s.filterSsoGroupsRegex != "" {
+		var filteredGroups []string
+		regex, err := regexp.Compile(s.filterSsoGroupsRegex)
+		if err != nil {
+			log.WithError(err).Errorf("failed to compile filterSsoGroupsRegex: %s", s.filterSsoGroupsRegex)
```

Comment:
I think it would make more sense to compile this on start-up and immediately error out then as opposed to erroring out much later when a request is made.